### ISSUE TITLE
Min MTU

### ISSUE THREAD

**Issue by BiagioFesta** [quinn-rs/quinn#1154]
Currently (`3f908a`), *quinn* uses [`const MIN_MTU: u16 = 1232;`](https://github.com/quinn-rs/quinn/blob/3f908a2c8c1ec4585212d776fafe536ea17bf2b4/quinn-proto/src/lib.rs#L313).

Maybe I am missing something, but looking at it, it seems this value is used more as *"datagram size"*.
Indeed, running the quinn server example (over IPv4):

* Client Invocation:
```
cargo run --example client https:/server:1234/README.md
```

* Server log:
```
[...]
Jul 06 02:59:45.746 TRACE drive{id=0}: quinn_proto::connection: sending 1232 bytes in 1 datagrams
[...]
```

The server sends a UDP datagram payload with 1232 bytes.

From my understanding, ([quic-draft-32 #14](https://datatracker.ietf.org/doc/html/draft-ietf-quic-transport-34#section-14)) the maximum UDP datagram payload should be set at `1200` bytes (unless afterwards increased by  PMTUD).

Is there any requirement for this design choice?

**Comment by djc** [comment]
Hmm, I think you're right. I find the wording in the spec a little confusing, since it discusses the maximum datagram size:

> The maximum datagram size is defined as the largest size of UDP payload that can be sent across a network path using a single UDP datagram. QUIC MUST NOT be used if the network path cannot support a maximum datagram size of at least 1200 bytes.

But then also goes on to discuss the minimum IP packet size:

> QUIC assumes a minimum IP packet size of at least 1280 bytes. This is the IPv6 minimum size [IPv6] and is also supported by most modern IPv4 networks. Assuming the minimum IP header size of 40 bytes for IPv6 and 20 bytes for IPv4 and a UDP header size of 8 bytes, this results in a maximum datagram size of 1232 bytes for IPv6 and 1252 bytes for IPv4. Thus, modern IPv4 and all IPv6 network paths are expected to be able to support QUIC.

And goes on to state:

> Any maximum datagram size larger than 1200 bytes can be discovered using Path Maximum Transmission Unit Discovery (PMTUD) (see Section 14.2.1) or Datagram Packetization Layer PMTU Discovery (DPLPMTUD) (see Section 14.3).

So we have three different things:

* IP packet size: minimum of 1280 for IPv6 and "most modern IPv4 networks"
* Subtracting minimum IP header sizes gives 1232 bytes for IPv6 and 1252 bytes for IPv4 to get a "maximum datagram size"
* But apparently we want to take into account for IP or UDP extensions

See also clarifications in https://github.com/quicwg/base-drafts/issues/4271.

I find the wording pretty confusing since it uses "maximum datagram size" to refer to what seems to be the UDP datagram payload size, whereas the UDP RFC talks about a datagram as being inclusive of the header (I find the latter more intuitive).

Anyway, I propose that we rename related values to `max_dgram_payload_len` or something to be explicit, and change the minimum to 1200 as proposed.

@BiagioFesta would you like to submit a PR?

**Comment by djc** [comment]
It looks like the lower minimum was introduced during draft 29, but I only found this commit:

https://github.com/quicwg/base-drafts/commit/0d20f5448aa08f89ae439e49bbeb9653aa62d5b3

I don't know how to find further discussion from here.

**Comment by Ralith** [comment]
That confusion in the spec has actually been bugging me for quite a while. The cited issue seems to suggest that the proposed change is correct, though. Thanks for digging that up!

**Comment by Ralith** [comment]
Fixed in #1156.
